### PR TITLE
Prepare to release 1.0

### PR DESCRIPTION
**Pull request by danizord**
- Update dependencies and make some depedencies explicit
- Add missing license headers
- Add ConfigProvider for - Add missing license headers

[BC break] I've renamed ModuleConfig to ConfigProvider for consistency with Zend\ComponentInstaller

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -17,14 +17,16 @@
...
         "php": "~7.0",
-        "guzzle/guzzle": "~3.6",
-        "guzzlehttp/guzzle": "~6.0",
-        "zendframework/zend-diactoros": "~1.1",
-        "container-interop/container-interop": "^1.1"
+        "container-interop/container-interop": "^1.1",
+        "guzzle/guzzle": "^3.9",
+        "guzzlehttp/guzzle": "^6.1",
+        "psr/http-message": "^1.0",
+        "symfony/event-dispatcher": "^2.8",
```
Comment (bakura10) [inline composer.json]: Why?

File: composer.json [hunk 2]
```diff
@@ -17,14 +17,16 @@
     ],
     "require": {
         "php": "~7.0",
-        "guzzle/guzzle": "~3.6",
-        "guzzlehttp/guzzle": "~6.0",
-        "zendframework/zend-diactoros": "~1.1",
-        "container-interop/container-interop": "^1.1"
+        "container-interop/container-interop": "^1.1",
+        "guzzle/guzzle": "^3.9",
+        "guzzlehttp/guzzle": "^6.1",
```
Comment (bakura10) [inline composer.json]: Why?

File: composer.json [hunk 3]
```diff
@@ -48,5 +50,12 @@
         "cs": "phpcs",
         "cs-fix": "phpcbf",
         "test": "phpunit"
+    },
+    "extra": {
+        "zf": {
```
Comment (bakura10) [inline composer.json]: what is that?

File: composer.json (hunk 2 above)
Comment (danizord) [inline composer.json]: We explicity use it in [TokenExchanger](https://github.com/zf-fr/zfr-shopify/blob/master/src/OAuth/TokenExchanger.php#L5)

Comment (bakura10) [inline composer.json]: Never mind, we are using Guzzle 6 for the OAuth methods.

File: composer.json (hunk 1 above)
Comment (danizord) [inline composer.json]: Explicity used in [ShopifyClient](https://github.com/zf-fr/zfr-shopify/blob/master/src/ShopifyClient.php#L25)

File: composer.json (hunk 3 above)
Comment (danizord) [inline composer.json]: Remember that ExpressiveConfig I had in mind? Mwop did it :D

File: composer.json (hunk 1 above)
Comment (bakura10) [inline composer.json]: But we do not instantiate any, it's a dependency from Guzzle 3.

Comment (danizord) [inline composer.json]: We [do](https://github.com/zf-fr/zfr-shopify/blob/master/src/ShopifyClient.php#L212).

File: composer.json (hunk 3 above)
Comment (bakura10) [inline composer.json]: So basically we no longer need to do that : https://github.com/maestrooo/shopify-sms-notifications/blob/master/php-app/config/config.php#L19 ?

**Comment by bakura10** [comment]
README must be updated as well.

File: composer.json (hunk 3 above)
Comment (danizord) [inline composer.json]: We still need it, but `Zend\ComponentInstaller` will automatically register our `ConfigProvider` there.

**Comment by bakura10** [comment]
This as well: https://github.com/zf-fr/zfr-shopify#installation

**Comment by bakura10** [comment]
Once you've updated the README completely, LGTM :).

**Comment by bakura10** [comment]
Still not enough :D. You need to update the version in the Installation section of README.

**Comment by bakura10** [comment]
Can tag !

**Pull request merged**